src/action/platform_info.rs

```diff
@@ -0,0 +1,236 @@
+    } 
+}
+
+#[cfg(target_os = "linux")]
+use libc::{uname, utsname};
+
+/// Function that returns `Response` for Linux operating systems
+fn get_linux_response<S: Session>(session: &mut S, os_type: String) -> session::Result<()> {
+    #[cfg(target_os = "linux")]
+    {
+        let linux_release_info = linux_os_release()
+                .map_err(Error::CannotGetLinuxRelease)?;
+
+        let mut system_info: utsname;
+
+        unsafe {
+            system_info = std::mem::zeroed();
+            uname(&mut system_info);
+        }
+
+        session.reply(Response {
+            system: Some(os_type),
+            release_name: linux_release_info.name,
+            version_id: linux_release_info.version_id,
+            machine: Some(convert_raw_string(&system_info.machine)),
+            kernel_release: Some(convert_raw_string(&system_info.release)),
+            fqdn: hostname().ok(),
+            architecture: Some(convert_raw_string(&system_info.machine)),
+            node: Some(convert_raw_string(&system_info.nodename))
+        })?;
+    }
+    Ok(())
+}
+
+/// Handles requests for `GetPlatformInfo` action.
+pub fn handle<S: Session>(session:&mut S, _: ()) -> session::Result<()> {
+    let os_type = os_type().map_err(Error::CannotGetOSType)?;
+    match os_type.as_str() {
+        "Linux" => {
+            get_linux_response(session, os_type)?;
+        },
+        _ => {
+            session.reply(Response {
+                    system: Some(os_type),
+                    fqdn: hostname().ok(),
+                    ..Default::default() // TODO: Add other systems
+                })?;
+        }
+    }
+
+    Ok(())
+}
+
+impl super::Response for Response {
+
+    const RDF_NAME: Option<&'static str> = Some("Uname");
+
+    type Proto = rrg_proto::Uname;
+
+    /// Convert `Response` struct to protobuf message `Uname`
+    fn into_proto(self) -> rrg_proto::Uname {
+        Uname {
+            system: self.system.clone(),
+            release: self.release_name.clone(),
+            version: self.version_id,
+            machine: self.machine,
+            kernel: self.kernel_release,
+            fqdn: self.fqdn,
+            architecture: self.architecture.clone(),
+            node: self.node,
+            pep425tag: Some(
+                format!("{}_{}_{}", 
+                    self.system.unwrap_or(String::from("None")), 
+                    self.release_name.unwrap_or(String::from("None")), 
+                    self.architecture.unwrap_or(String::from("None"))
+            )),
+            ..Default::default()
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    use super::*;
+
+    #[test]
+    fn test_system() {
+        let mut session = session::test::Fake::new();
+        assert!(handle(&mut session, ()).is_ok());
+
+        assert_eq!(session.reply_count(), 1);
+        let platform_info = &session.reply::<Response>(0);
+
+        assert!(platform_info.system.is_some());
+    }
```

Comment:
For this test you could actually have platform-specific test cases for every supported platform (through `#[cfg_attr(..., ignore)]`) that actually test this field for some specific values (e.g. "Linux").